app/lib/actions/katello/content_view/publish.rb

```diff
@@ -40,11 +40,22 @@ def plan(content_view, description = "", options = {})
             plan_action(ContentView::AddToEnvironment, version, library)
             repository_mapping = plan_action(ContentViewVersion::CreateRepos, version, source_repositories).repository_mapping
 
+            # Split Pulp 3 Yum repos out of the repository_mapping.  Only Pulp 3 RPM plugin has multi repo copy support.
+            separated_repo_map = separated_repo_mapping(repository_mapping, content_view)
+
+            if separated_repo_map[:pulp3_yum_depsolve].keys.flatten.present? &&
+                SmartProxy.pulp_master.pulp3_support?(separated_repo_map[:pulp3_yum_depsolve].keys.flatten.first) &&
+                content_view.solve_dependencies
+              plan_action(Repository::MultiCloneToVersion, separated_repo_map[:pulp3_yum_depsolve], version)
```

Comment:
thinking ahead to when pulp2 is gone, would it simplify things to always use MultiCloneToVersion even if solve_dependencies is false?  I assume the multi-repo copy api supports turning off dep solving?

Reply:
thinking logically, it then becomes:

if pulp3 supports yum, use MultiCloneToVersion for yum repos, otherwise use CloneToVersion

Reply:
Yeah that definitely makes sense to me, when writing this I knew that we'd want to use MultiCloneToVersion without dep solving too at some point.  I didn't include yet because I was worried about taking too much time getting this PR into 3.16, but I can fix it if we do have time.  The potentially time-consuming bit would just be getting the composite content view functionality right.

Reply:
I started implementing the changes to use MultiCloneToVersion for all Pulp 3 Yum content, and it's been working well so far.  I need to do a little testing, but I've been able to get composite content views to "merge" pretty well.

Reply:
I pushed a new commit that has my work so far.

Reply:
Looks like the only part of this that isn't working right is incremental update and `--propagate-all-composites true`.  It does work if I simply publish the composite, though.